File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        //here we need to make another vector array which will track the seq length per number
        
        vector<int> res(nums.size(),1);

        //looping
        for (int i = 0; i<nums.size();i++){
            for (int j = 0; j<i;j++){
                //i is the number we are considering, j will give us numbers that are prev from it
                if (nums[j]<nums[i]){
                    res[i] = max(res[i],res[j]+1);
                }
            }
        };

        return *max_element(res.begin(),res.end());
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit=0
        curr_min=prices[0]
        for i in prices:
            if i>curr_min:
                max_profit=max(max_profit,i-curr_min)
            else:
                curr_min=i
        return max_profit
        